plugins/inputs/ethtool/ethtool.go

```diff
@@ -48,14 +71,310 @@ type Ethtool struct {
 	command Command
 }
 
+type CommandEthtool struct {
+	Log                 telegraf.Logger
+	namespaceGoroutines map[string]*NamespaceGoroutine
+}
+
 func (*Ethtool) SampleConfig() string {
 	return sampleConfig
 }
 
-const (
-	pluginName       = "ethtool"
-	tagInterface     = "interface"
-	tagNamespace     = "namespace"
-	tagDriverName    = "driver"
-	fieldInterfaceUp = "interface_up"
-)
```

Comment:
> Instead of completely changing the setup you should just move those definitions into the `_linux.go` file!

@srebhan Can you elaborate how would this help with following findings for `non-linux` OSes?
```
plugins/inputs/ethtool/ethtool.go:43:2                      unused  field `interfaceFilter` is unused
plugins/inputs/ethtool/ethtool.go:44:2                      unused  field `namespaceFilter` is unused
plugins/inputs/ethtool/ethtool.go:45:2                      unused  field `includeNamespaces` is unused
plugins/inputs/ethtool/ethtool.go:48:2                      unused  field `command` is unused
```

I strongly believe that situation in `ethtool` plugin is much clearer right now. Content of `ethtool.go` and `ethtool_linux.go` were merged into single `ethtool.go` for `linux` only. Other OSes are gently handled by `ethtool_notlinux.go`.

Reply:
The reason why I don't like the structural changes is
1. It is a massive change not necessarily required to accommodate the linter issue.
2. The structure is now again different from other plugins that are platform dependent!

> Can you elaborate how would this help with following findings for non-linux OSes?

Those constants are used in the linux-only build and are undefined in the non-linux build. Doesn't that solve the linter issues?

Reply:
I must admit that I don't know why you are talking about those constants. Problem is only with struct fields:
```
plugins/inputs/ethtool/ethtool.go:43:2     unused  field `interfaceFilter` is unused
plugins/inputs/ethtool/ethtool.go:44:2     unused  field `namespaceFilter` is unused
plugins/inputs/ethtool/ethtool.go:45:2     unused  field `includeNamespaces` is unused
plugins/inputs/ethtool/ethtool.go:48:2     unused  field `command` is unused
```

I think that we need to have dedicated talk about best solution. For now, I'm reverting changes for `ethtool` plugin.